Added info on new Java features (#342)
paketo-buildpacks/paketo-website: content/docs/howto/java.md

## Enable Remote Debugging

If `BPL_DEBUG_ENABLED` is set to `true` at runtime the application will be configured to accept debugger connections. The debug port defaults to `8000` and can be configured with `BPL_DEBUG_PORT` at runtime. If `BPL_DEBUG_SUSPEND` is set at runtime, the JVM will suspend execution until a debugger has attached.

**Example**: Remote Debugging

> To run an image with the debug port published:

Reviewer: ```suggestion
The following command builds a debug-enabled image.

{{< code/copyable >}}
pack build samples/java \
  --path java/jar
{{< /code/copyable >}}

To run an image with the debug port published:
```

I think we should still leave in the build command even though there's nothing specific that needs to be done. It just makes the how-to feel more complete.

## Enable JMX

If `BPL_JMX_ENABLED` is set at runtime, the application will be configured to accept [JMX][jmx] connections. The JMX connector will listen on port `5000` by default. The port can be configured with the `BPL_JMX_PORT` environment variable at runtime.

**Example**: Enabling JMX

> To run an image with the JMX port published:

Reviewer: ```suggestion
The following command builds a JMX-enabled image.

{{< code/copyable >}}
pack build samples/java \
  --path java/jar 
{{< /code/copyable >}}

To run an image with the JMX port published:
```

Similar to my previous comment. This just seems more complete.

{{< code/copyable >}}
docker run --env BPL_JMX_ENABLED=true --publish 5000:5000 samples/java
{{< /code/copyable >}}

Connect [JConsole][jconsole] to the published port.
![JConsole](/images/jconsole.png)
  
## Enable Java Native Memory Tracking (NMT)

> By default, the JVM will be configured track internal memory usage. The JVM will print its last memory usage data when it exits, the level of detail can be configured at runtime by setting the environment variable `BPL_JAVA_NMT_LEVEL`, which supports both `summary` (default) and `detail`. Since there is a small amount of overhead required to support NMT, it can be disabled by setting the environment variable `BPL_JAVA_NMT_ENABLED` to `false`.

Reviewer: ```suggestion
By default, the JVM will be configured to track internal memory usage. The JVM will print its last memory usage data when it exits, the level of detail can be configured at runtime by setting the environment variable `BPL_JAVA_NMT_LEVEL`, which supports both `summary` (default) and `detail`. Since there is a small amount of overhead required to support NMT, it can be disabled by setting the environment variable `BPL_JAVA_NMT_ENABLED` to `false`.
```

  
**Example**: Capturing NMT output

To capture NMT data using the JDK tool `jcmd`, first ensure that you have a JDK installed at runtime (see [Install a Specific JVM Type][install-jvm-type]). Then run the following from within the container: 

{{< code/copyable >}}

> jcmd 1 VM.native_memory summary

Reviewer: Did you try this? Will `jcmd` be on the path?

I feel like we might want to add the `docker exec` command cause I think you'd need the launcher so that $PATH is set up correctly, or you'd need the full path  to the executable.

Author: Resolved with docker exec command addition

{{< /code/copyable >}}

The first argument should be the JVM PID, in the case of the Paketo Java buildpack, this will be `1`.

## Enable Java Flight Recorder (JFR)

> If `BPL_JFR_ENABLED` is set to `true` at runtime, Java Flight Recording features will be enabled by the JVM. To configure JFR via it's supported arguments, add them to the optional environment variable `BPL_JFR_ARGS` at runtime.

Reviewer: ```suggestion
If `BPL_JFR_ENABLED` is set to `true` at runtime, Java Flight Recording features will be enabled by the JVM. To configure JFR via [its supported arguments](https://docs.oracle.com/javacomponents/jmc-5-4/jfr-runtime-guide/run.htm#JFRUH178), add them to the optional environment variable `BPL_JFR_ARGS` at runtime.
```

Two default arguments are configured for JFR as follows:

* `dumponexit` - this is set to `true` to write the recording data to a file when the JVM exits
* `filename` - this is set to `<system-temp-dir>/recording.jfr` (i.e. `/tmp/recording.jfr`) and is the location where the recording data will be written on JVM exit

If any args are provided via `BPL_JFR_ARGS`, these defaults will not be configured. 

**Example**: Enabling & configuring JFR

> To run an image with JFR enabled and optionally configure it with custom arguments:

Reviewer: ```suggestion
The following command builds a JFR-enabled image.

{{< code/copyable >}}
pack build samples/java \
  --path java/jar
{{< /code/copyable >}}

To run an image with JFR enabled and optionally configure it with custom arguments:
```

Again, makes the example more complete.
